src/vsphere_cpi/lib/cloud/vsphere/storage_list.rb

```diff
@@ -0,0 +1,28 @@
+module VSphereCloud
+  module StorageList
+
+    # storageList is a list of datastores and datastore_clusters defined in cloud_properties as 'datastores'
+    # Eg. [datastore1, clusters: [{datastore_cluster1: {}, datatore_cluster2: {}}]]
+
+    #Returns list of datastore names
+    def datastore_names
+      storage_list.select do |entry|
+        entry.is_a?(String)
+      end
+    end
+
+    #Returns list of StoragePod objects for datastore_clusters
+    def datastore_clusters
+      clusters_hash = storage_list.find do |entry|
+        hash = Hash.try_convert(entry)
+        next if hash.nil?
+        hash.key?('clusters')
+      end
+      return [] if clusters_hash.nil?
```

Comment:
This is one of those rare cases where an `or return []` may be justified at the end of the previous line. How do we feel about `and`/`or` in Ruby?